types/geojson/index.d.ts

```diff
@@ -68,6 +68,7 @@ export interface GeoJsonObject {
  */
 export interface GeometryObject extends GeoJsonObject {
     type: GeoJsonGeometryTypes;
+    coordinates?: Position | Position[] | Position[][] | Position[][][];
```

Comment:
A `GeometryCollection` is also a `GeometryObject`, but is not allowed to have a `coordinates` property. At a first glance it seems to be incorrect for me.

Why do you use it here? Maybe we will find another, better solution...

Reply:
but if the geometry type (`GeoJsonGeometryTypes`) is `Point` or `Polygon` it can have coordinates no? That's why I made it optional, in case it is a `GeometryCollection ` and shouldn't have a coordinates

Reply:
I think I got the point. But i would not prefer to make it like this. I take a look at an [older change](https://github.com/DefinitelyTyped/DefinitelyTyped/commit/89667eb5cbee013f697ef1adef6582caa62385f3#diff-4f5aa92a196385006e0183c771a95f12L33) especially on the old line 33. In my opinion the `GeometryObject` should be a `type` again, like this:

```typescript
export type GeometryObject = Point | /* other geometries */ | GeometryCollection
// with Geometries like this:

export interface Point extends GeoJsonObject {
    type: "Point";
    coordinates: Position;
}
// and so on...
```

In short, I think there is no advantage to extends geometries from the `GeometryObject` instead of `GeoJsonObject` directly.

What do @cobster and @JeffJacobson think about that?